src/gc-stock.c

```diff
@@ -3508,6 +3553,11 @@ void jl_parallel_gc_threadfun(void *arg)
         uv_mutex_unlock(&gc_threads_lock);
         assert(jl_atomic_load_relaxed(&ptls->gc_state) == JL_GC_PARALLEL_COLLECTOR_THREAD);
         gc_mark_loop_parallel(ptls, 0);
+        if (may_sweep_stack(ptls)) {
+            assert(jl_atomic_load_relaxed(&ptls->gc_state) == JL_GC_PARALLEL_COLLECTOR_THREAD);
+            sweep_stack_pool_loop();
+            jl_atomic_fetch_add(&ptls->gc_tls.gc_stack_sweep_requested, -1);
+        }
```

Comment:
You should probably have a `!may_sweep_stack(...)` in the condition of the `while` loop that encloses the `cond_wait`.

I.e.

`while (!may_mark() && !may_sweep(ptls) && !may_sweep_stack(ptls))`